pkg/apis/batch/types.go

```diff
@@ -340,6 +352,19 @@ type JobSpec struct {
 	//
 	// +optional
 	Suspend *bool
+
+	// podReplacementPolicy specifies when to create replacement Pods.
+	// Possible values are:
+	// - TerminatingOrFailed means to create a replacement Pod when the previously
+	//   created Pod is terminating or failed.
```

Comment:
should this mention terminated/succeeded as well?

the failed has the same semantic as in `Failed` where we mention it, right?

```
	// - Failed means to wait until a previously created Pod is fully terminated (has phase
	//   Failed or Succeeded) before creating a replacement Pod.
```


It might be good to also make a short explanation about the problem with succeeded counting against failed: https://github.com/kubernetes/enhancements/blob/master/keps/sig-apps/3939-allow-replacement-when-fully-terminated/README.md#the-default-job-controller-behavior

Reply:
I think it's too much detail for the API comment. But we should definitely leave a note in the documentation.

Reply:
I'll add that to the my list in the doc.